exec/shell.go

```diff
@@ -13,16 +13,20 @@ type Executor interface {
 	Execute(name string, args ...string) (string, error)
 	// ExecuteStdout executes a command and prints the standard output instead of returning it.
 	ExecuteStdout(name string, args ...string) error
+	// ExecuteWithBuffers executes a command and forwards stdout and stderr
+	ExecuteWithBuffers(stdout, stderr *bytes.Buffer, name string, args ...string) error
```

Comment:
Use the `io.Writer` interface instead, bytes.Buffer implement it, as well as numerous other implementations (i.e `os.File`).

```suggestion
	ExecuteWithBuffers(stdout, stderr io.Writer, name string, args ...string) error
```
The name of the method may have to be changed as well